File: datasets/cpp-keep-code_davinci_001_temp_0.2/HumanEval_136_largest_smallest_integers.cpp
#include<assert.h>
#include<bits/stdc++.h>
// Create a function that returns a tuple (a, b), where 'a' is
// the largest of negative integers, and 'b' is the smallest
// of positive integers in a list.
// If there is no negative or positive integers, return them as None.
// Examples:
// largest_smallest_integers([2, 4, 1, 3, 5, 7]) == (None, 1)
// largest_smallest_integers([]) == (None, None)
// largest_smallest_integers([0]) == (None, None)
std::tuple<std::optional<long>, std::optional<long>> largest_smallest_integers(std::vector<long> lst) {
    if (lst.empty()) {
        return {std::nullopt, std::nullopt};
    }
    long max = std::numeric_limits<long>::min();
    long min = std::numeric_limits<long>::max();
    for (auto i : lst) {
        if (i < 0) {
            max = std::max(max, i);
        } else {
            min = std::min(min, i);
        }
    }
    return {max, min};

}
int main() {
    auto candidate = largest_smallest_integers;
    assert(candidate((std::vector<long>({(long)2, (long)4, (long)1, (long)3, (long)5, (long)7}))) == std::make_tuple(std::optional<long>({}), std::optional<long>(1)));
    assert(candidate((std::vector<long>({(long)2, (long)4, (long)1, (long)3, (long)5, (long)7, (long)0}))) == std::make_tuple(std::optional<long>({}), std::optional<long>(1)));
    assert(candidate((std::vector<long>({(long)1, (long)3, (long)2, (long)4, (long)5, (long)6, (long)-2}))) == std::make_tuple(-2, 1));
    assert(candidate((std::vector<long>({(long)4, (long)5, (long)3, (long)6, (long)2, (long)7, (long)-7}))) == std::make_tuple(-7, 2));
    assert(candidate((std::vector<long>({(long)7, (long)3, (long)8, (long)4, (long)9, (long)2, (long)5, (long)-9}))) == std::make_tuple(-9, 2));
    assert(candidate((std::vector<long>())) == std::make_tuple(std::optional<long>({}), std::optional<long>({})));
    assert(candidate((std::vector<long>({(long)0}))) == std::make_tuple(std::optional<long>({}), std::optional<long>({})));
    assert(candidate((std::vector<long>({(long)-1, (long)-3, (long)-5, (long)-6}))) == std::make_tuple(std::optional<long>(-1), std::optional<long>({})));
    assert(candidate((std::vector<long>({(long)-1, (long)-3, (long)-5, (long)-6, (long)0}))) == std::make_tuple(std::optional<long>(-1), std::optional<long>({})));
    assert(candidate((std::vector<long>({(long)-6, (long)-4, (long)-4, (long)-3, (long)1}))) == std::make_tuple(-3, 1));
    assert(candidate((std::vector<long>({(long)-6, (long)-4, (long)-4, (long)-3, (long)-100, (long)1}))) == std::make_tuple(-3, 1));
}

File: datasets/js-keep-code_davinci_001_temp_0.2/HumanEval_136_largest_smallest_integers.js
//Create a function that returns a tuple (a, b), where 'a' is
// the largest of negative integers, and 'b' is the smallest
// of positive integers in a list.
// If there is no negative or positive integers, return them as None.
// Examples:
// largest_smallest_integers([2, 4, 1, 3, 5, 7]) == (None, 1)
// largest_smallest_integers([]) == (None, None)
// largest_smallest_integers([0]) == (None, None)
function largest_smallest_integers(lst){
    var largest = null;
    var smallest = null;
    for (var i = 0; i < lst.length; i++) {
        if (lst[i] < 0) {
            if (largest == null || lst[i] > largest) {
                largest = lst[i];
            }
        } else {
            if (smallest == null || lst[i] < smallest) {
                smallest = lst[i];
            }
        }
    }
    return [largest, smallest];
}


const assert = require('node:assert');


function test() {
  let candidate = largest_smallest_integers;
  assert.deepEqual(candidate([2, 4, 1, 3, 5, 7]),[null, 1]);
  assert.deepEqual(candidate([2, 4, 1, 3, 5, 7, 0]),[null, 1]);
  assert.deepEqual(candidate([1, 3, 2, 4, 5, 6, -2]),[-2, 1]);
  assert.deepEqual(candidate([4, 5, 3, 6, 2, 7, -7]),[-7, 2]);
  assert.deepEqual(candidate([7, 3, 8, 4, 9, 2, 5, -9]),[-9, 2]);
  assert.deepEqual(candidate([]),[null, null]);
  assert.deepEqual(candidate([0]),[null, null]);
  assert.deepEqual(candidate([-1, -3, -5, -6]),[-1, null]);
  assert.deepEqual(candidate([-1, -3, -5, -6, 0]),[-1, null]);
  assert.deepEqual(candidate([-6, -4, -4, -3, 1]),[-3, 1]);
  assert.deepEqual(candidate([-6, -4, -4, -3, -100, 1]),[-3, 1]);
}

test();